src/pdm/utils.py

```diff
@@ -410,3 +410,8 @@ def is_pip_compatible_with_python(python_version: Version | str) -> bool:
     pip = importlib_metadata.distribution("pip")
     requires_python = get_specifier(pip.metadata["Requires-Python"])
     return requires_python.contains(python_version, True)
+
+
+def path_without_fragments(path: str) -> Path:
+    """Remove egg fragment from path"""
+    return Path(path.split("#")[0])
```

Comment:
Should you `.rsplit("#", 1)[0]` to support hashes in the file path? Does that make sense?

Reply:
In this case how would we know the hash is on the file path or is a end real fragment ?

Reply:
Fixed with commit https://github.com/pdm-project/pdm/pull/1552/commits/8a361c9c8c3ea91b18f05d2a46d5585ca44c41f0